### PR TITLE
Adds: API js injection example.

### PR DESCRIPTION
**Pull request by korvyashka**
Adds API js injection example.

Summary:

Adds API js injection example.

It shows how having a screen,
create an asset, patch it with js_injection and attach to screen using newly created playlist.

While doing example, found several API and one player issue, that should be fixed later - will create tasks,
but this API example will work regardless.

Test Plan:
No need for test plan.

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+#### Code example
+
+```python
+
+import requests
+from time import sleep
+
+TOKEN = '<API TOKEN>'
```
Comment (vpetersson) [inline api/README.md:19]: use `os.getenv()` instead.

File: api/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+SCREEN_ID = '<Screen id>'
```
Comment (vpetersson) [inline api/README.md:20]: Same.

File: api/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+### Summary
+
+This example shows how, having a screen, attach to it a newly created asset with js injection.
+Here are some examples and tips on writing a proper js injection code: * [Js Injection](../javascript-injectors/README.md)
```
Comment (vpetersson) [inline api/README.md:4]: Either write out JavaScript fully, or abbreviate it JS.

File: api/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+For this you would need an API token(Can be found on web dashboard /manage/account/team)
```
Comment (vpetersson) [inline api/README.md:6]: This is an invalid sentence. Please run these sentences through [Grammarly](https://www.grammarly.com/) or similar.

**Review by vpetersson — CHANGES_REQUESTED** [review]
Can we rewrite this more as a tutorial than just a bunch of code?

**Comment by vpetersson** [comment]
@korvyashka i've enabled the domain `playground.srly.io` for this project. We now need to hook it up with Pages to serve it.

Here's an example of how you do that:

```
# Simple workflow for deploying static content to GitHub Pages
name: Deploy static content to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["master"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow one concurrent deployment
concurrency:
  group: "pages"
  cancel-in-progress: true

jobs:
  # Single deploy job since we're just deploying
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Setup Pages
        uses: actions/configure-pages@v2
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
        with:
          # Upload entire repository
          path: '.'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```

**Comment by vpetersson** [comment]
Take a look at some of stripes tutorials for inspiration 

https://stripe.com/docs/development/get-started

**Review by renatgalimov — APPROVED** [review]

File: api/README.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
+### Summary
+
+This example shows how, having a screen, attach to it a newly created asset with js injection.
```
Comment (vpetersson) [inline api/README.md:3]: Please rewrite this sentence.

File: README.md [hunk 6]
```diff
@@ -9,3 +9,4 @@ This repository holds various examples that showcases Screenly's features and AP
 * [JavaScript Injectors](https://github.com/Screenly/playground/tree/master/javascript-injectors/)
 * [Weather App](https://github.com/Screenly/weather-app)
 * [Clock App](https://github.com/Screenly/clock-app)
+* [API](./api/README.md)
```
Comment (vpetersson) [inline README.md:12]: it's not an API example. It's really a demo of the JavaScript injector, so it probably rather belongs as part of that.

File: .github/workflows/pages.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "master" branch
+  push:
+    branches: [ "feature/api_js_injection_example" ]
```
Comment (vpetersson) [inline .github/workflows/pages.yml:8]: Needs to be changed to `master` before merging.

File: api/README.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+For this, you would need
+ - an API token, which can be found on the web dashboard /manage/account/team.
```
Comment (vpetersson) [inline api/README.md:6]: Capitalize.

File: api/README.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
+### Summary
+
+This example shows how, having a screen, attach to it a newly created asset with js injection.
+
+For this, you would need
```
Comment (vpetersson) [inline api/README.md:5]: Missing colon.

File: api/README.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+ - an API token, which can be found on the web dashboard /manage/account/team.
+ - a paired screen id, which can be found either from the API or retrieved from the web dashboard screen URL:
```
Comment (vpetersson) [inline api/README.md:7]: Capitalize.

File: api/README.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+https://yourteam.screenlyapp.com/manage/screens/01FAZMW8E0CB2AM78N63QP406Y
```
Comment (vpetersson) [inline api/README.md:9]: We should use the API for this. We don't want people to have to manually parse URLs to dig out data.

File: api/README.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+TOKEN = os.getenv('TOKEN')
+
+HEADERS = {
+    "Authorization": f"Token {TOKEN}",
+    "Content-Type": "application/json"
+}
+```
+
+Then, let's create a web asset we want to show on the screen.\
+Web asset also will have a js injection - js code, that will run once the page is loaded.\
```
Comment (vpetersson) [inline api/README.md:31]: js -> JS. Also, please rewrite this sentence.

**Review by vpetersson — CHANGES_REQUESTED** [review]

File: api/README.md [hunk 13]
```diff
@@ -1,21 +1,44 @@
...
- - a paired screen id, which can be found either from the API or retrieved from the web dashboard screen URL:
+Any request to the API requires an API token. \
+You can acquire it in Screenly [Web dashboard](https://your-domain.screenlyapp.com/manage/account/team) by adding a new token.
 
-https://yourteam.screenlyapp.com/manage/screens/01FAZMW8E0CB2AM78N63QP406Y
+![](./img/create_token.png)
 
-'01FAZMW8E0CB2AM78N63QP406Y' - this is screen_id.
+To make API requests tutorial uses python 3 and python [requests](https://pypi.org/project/requests/) library to make API calls.
+You can use curl or something else alternatively.
```
Comment (vpetersson) [inline api/README.md:11]: curl -> `curl`.

File: api/README.md [hunk 14]
```diff
@@ -1,21 +1,44 @@
...
- - an API token, which can be found on the web dashboard /manage/account/team.
- - a paired screen id, which can be found either from the API or retrieved from the web dashboard screen URL:
+Any request to the API requires an API token. \
+You can acquire it in Screenly [Web dashboard](https://your-domain.screenlyapp.com/manage/account/team) by adding a new token.
 
-https://yourteam.screenlyapp.com/manage/screens/01FAZMW8E0CB2AM78N63QP406Y
+![](./img/create_token.png)
 
-'01FAZMW8E0CB2AM78N63QP406Y' - this is screen_id.
+To make API requests tutorial uses python 3 and python [requests](https://pypi.org/project/requests/) library to make API calls.
```
Comment (vpetersson) [inline api/README.md:10]: python -> Python

File: README.md (hunk 6 above)
Comment (korvyashka) [reply README.md:12]: Actually, I think it is an API example more than JS injection example - JS injection is just a convenient web asset - we could use any other asset type in this example.

File: api/README.md [hunk 15]
```diff
@@ -27,8 +50,42 @@ HEADERS = {
...
+# For an asset to be associated with the screen we need to label the playlist and screen with the same group.
+group_id = create_group(SCREEN_ID, "My Js injection Group")
+```
+
+#### Create Web Asset
+
+[API](https://developer.screenlyapp.com/#operation/assets_create)
+
+Here, we are creating a web asset with JS injection.
+JS injection is a javascript code, that will run once a page is loaded on your screen.
```
Comment (vpetersson) [inline api/README.md:87]: javascript -> JavaScript.

File: api/README.md [hunk 16]
```diff
@@ -27,8 +50,42 @@ HEADERS = {
...
+
 Here are some examples and tips on writing a proper JS injection code: [Js Injection](../javascript-injectors/README.md)
```
Comment (vpetersson) [inline api/README.md:90]: JS Injector

**Review by vpetersson — CHANGES_REQUESTED** [review]
Added some minor feedback, but more importantly, how will this be served? We are not doing any processing of the `README.md` at all. From what I gather, all this flow will do is to copy everything in `.` to a static bucket. It won't actually process this at all and generate a page, like `playground.srly.io/javascript-example/index.html` or similar.

File: api/README.md [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+
+```python
+import requests
+
+
+# This js injection will be applied when page is shown on the device.
+JS_CODE = """
+document.getElementById("hello-world").textContent = 'New Text';
+"""
+SOURCE_URL = 'https://screenly.github.io/screenly-playground.github.io/'
```
Comment (vpetersson) [inline api/README.md:100]: This should be `https://playground.srly.io`.

File: .github/workflows/pages.yml [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./api/pages/
+          destination: ./_site
```
Comment (vpetersson) [inline .github/workflows/pages.yml:36]: Can you push this to a sub-folder instead? We want other workflows to also be able to push here.

**Review by vpetersson — CHANGES_REQUESTED** [review]

**Review by vpetersson — APPROVED** [review]

**Pull request merged**